langchain/agents/agent.py

```diff
@@ -419,7 +427,15 @@ async def aplan(
             Action specifying what tool to use.
         """
         full_inputs = self.get_full_inputs(intermediate_steps, **kwargs)
-        action = await self._aget_next_action(full_inputs)
+        try:
+            action = await self._aget_next_action(full_inputs)
+        except ValueError as e:
+            if "parse" in str(e):
```

Comment:
There's the the `OutputParserException` type that should be used instead of matching on the presence of a string. Don't want to match "UnitError: parsec not valid distance" or something

One could argue that the agent should recover from any error of its own accord, but that can also be handled in the output parser itself, and I'm not sure I'd want to make that the default

Reply:
Refactored agents to use that exception instead of value error